### PR TITLE
Runtime for new Kusama Parachain

### LINKED ISSUE #580: Runtime for new Kusama Parachain
- [x] Reserve new para id
- [x] Prepare runtime with new id and secrets. Can be anything as long as it can produce blocks and has sudo
- [x] Get genesis state and wasm blob of new runtime.
- [x] Launch nodes for new parachain
- [x] Win auction
- [x] Start producing blocks
- [x] Send pallet_xcm.send(Transact(regisrar.swap(new_id, 2084)) from the new parachain to Kusama, with sudo.
- [x] Send pallet_xcm.send(Transact(regisrar.swap(2084, new_id)) from Calamari to Kusama, with governance.

### PR DESCRIPTION
**Pull request by ghzlatarev**
## Description

closes: #580

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Linked to Github issue with discussion and accepted design OR have an explanation in the PR that describes this work.
- [ ] Wrote unit tests.
- [ ] Updated relevant documentation in the code.
- [ ] Added **one** line describing your change in `<branch>/CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer.
- [ ] If runtime changes, need to update the version numbers properly:
   * `authoring_version`: The version of the authorship interface. An authoring node will not attempt to author blocks unless this is equal to its native runtime.
   * `spec_version`: The version of the runtime specification. A full node will not attempt to use its native runtime in substitute for the on-chain Wasm runtime unless all of spec_name, spec_version, and authoring_version are the same between Wasm and native.
   * `impl_version`: The version of the implementation of the specification. Nodes are free to ignore this; it serves only as an indication that the code is different; as long as the other two versions are the same then while the actual code may be different, it is nonetheless required to do the same thing. Non-consensus-breaking optimizations are about the only changes that could be made which would result in only the impl_version changing.
   * `transaction_version`: The version of the extrinsics interface. This number must be updated in the following circumstances: extrinsic parameters (number, order, or types) have been changed; extrinsics or pallets have been removed; or the pallet order in the construct_runtime! macro or extrinsic order in a pallet has been changed. You can run the `metadata_diff.yml` workflow for help. If this number is updated, then the `spec_version` must also be updated 
- [ ] Verify benchmarks & weights have been updated for any modified runtime logics
- [ ] If importing a new pallet, choose a proper module index for it, and allow it in `BaseFilter`. Ensure **every** extrinsic works from front-end. If there's corresponding tool, ensure both work for each other.
- [ ] If needed, update our Javascript/Typescript APIs. These APIs are officially used by exchanges or community developers.
- [ ] If modifying existing runtime storage items, make sure to implement storage migrations for the runtime and test them with `try-runtime`. This includes migrations inherited from upstream changes, and you can search the diffs for modifications of `#[pallet::storage]` items to check for any.

### CODE REVIEW

File: node/src/command.rs [hunk 1]
```diff
@@ -40,7 +40,7 @@ pub type Block = generic::Block<Header, OpaqueExtrinsic>;
 
 pub const MANTA_PARACHAIN_ID: u32 = 2015;
 pub const CALAMARI_PARACHAIN_ID: u32 = 2084;
-pub const DOLPHIN_PARACHAIN_ID: u32 = 2084;
+pub const DOLPHIN_PARACHAIN_ID: u32 = 2085;
```
Comment (ghzlatarev) [inline node/src/command.rs]: TODO: change to actual reserved id.

File: node/src/chain_specs/dolphin.rs [hunk 2]
```diff
@@ -216,10 +216,145 @@ fn dolphin_dev_genesis(
...
+			// account id: dmtwRyEeNyRW3KApnTxjHahWCjN5b9gDjdvxpizHt6E9zYkXj
+			hex!["0027131c176c0d19a2a5cc475ecc657f936085912b846839319249e700f37e79"].into(),
+			hex!["8ebf03bda1702d719f428bc0a4c7cfca010c44a48ef79752490818c901548d20"]
+				.unchecked_into(),
+		),
+	];
+
+	// root account: dmyBqgFxMPZs1wKz8vFjv7nD4RBu4HeYhZTsGxSDU1wXQV15R
+	let root_key: AccountId =
+		hex!["bc153ffd4c96de7496df009c6f4ecde6f95bf67b60e0c1025a7552d0b6926e04"].into();
```
Comment (ghzlatarev) [inline node/src/chain_specs/dolphin.rs]: TODO: change to whatever sudo account is going to be used.

**Comment by Dengjianping** [comment]
1. ~~Can we use manta instead of dolphin for auction? I mean dolphin is kind of heavy to do such job.~~
2. 3 collators should be enough.
3. Launch nodes more early to catch blocks. Perhaps there's snapshot which can accelerate the sync..
4. I didn't see new new peer ids. Need to be created as well.
5. Related names like the name of runtime, token symbol, etc.

**Comment by ghzlatarev** [comment]
> 1. ~Can we use manta instead of dolphin for auction? I mean dolphin is kind of heavy to do such job.~
> 2. 3 collators should be enough.
> 3. Launch nodes more early to catch blocks. Perhaps there's snapshot which can accelerate the sync..
> 4. I didn't see new new peer ids. Need to be created as well.
> 5. Related names like the name of runtime, token symbol, etc.

We don't need 5. Just need to use different ID.

**Review by Garandor — COMMENTED** [review]
LGTdeploy (we're not merging this to manta, right?)

**Comment by ghzlatarev** [comment]
> LGTdeploy (we're not merging this to manta, right?)

nope.

**Pull request closed without merging**